heat/core/operations.py

```diff
@@ -11,6 +11,127 @@
 
 __all__ = []
 
+def __binary_bit_op(method, t1, t2):
+    """
+    Generic wrapper for element-wise binary bit-wise operations of two operands (either can be tensor or scalar).
+    Takes the method string for the operation function call in the torch tensor and the two operands involved in the operation as arguments.
+
+    Parameters
+    ----------
+    method : string
+        The operation to be performed. String that contains the method name in torch.tensor, which performs bit-wise operation elements-wise on the involved tensors,
+        e.g. add values from other to self
+
+    t1: dndarray or scalar
+        The first operand involved in the operation, must be integer or boolean
+
+    t2: tensor or scalar
+        The second operand involved in the operation, must be integer or boolean
+
+    Returns
+    -------
+    result: ht.DNDarray
+        A tensor containing the element-wise results of bit-wise operation.
+    """
+    if np.isscalar(t1):
+        try:
+            t1 = factories.array([t1])
+        except (ValueError, TypeError,):
+            raise TypeError('Data type not supported, input was {}'.format(type(t1)))
+
+        if np.isscalar(t2):
+            try:
+                t2 = factories.array([t2])
+            except (ValueError, TypeError,):
+                raise TypeError('Only numeric scalars are supported, but input was {}'.format(type(t2)))
+            output_shape = (1,)
+            output_split = None
+            output_device = None
+            output_comm = MPI_WORLD
+        elif isinstance(t2, dndarray.DNDarray):
+            output_shape = t2.shape
+            output_split = t2.split
+            output_device = t2.device
+            output_comm = t2.comm
+        else:
+            raise TypeError('Only tensors and numeric scalars are supported, but input was {}'.format(type(t2)))
+
+        if t1.dtype != t2.dtype:
+            raise TypeError('Tensors must have the same element type, but inputs were {} and {}'.format(t1.dtype,t2.dtype))
+
+    elif isinstance(t1, dndarray.DNDarray):
+        if np.isscalar(t2):
+            try:
+                t2 = factories.array([t2])
+                output_shape = t1.shape
+                output_split = t1.split
+                output_device = t1.device
+                output_comm = t1.comm
+            except (ValueError, TypeError,):
+                raise TypeError('Data type not supported, input was {}'.format(type(t2)))
+
+        elif isinstance(t2, dndarray.DNDarray):
+            if t1.split is None:
+                t1 = factories.array(t1, split=t2.split, copy=False, comm=t1.comm, device=t1.device, ndmin=-t2.numdims)
+            elif t2.split is None:
+                t2 = factories.array(t2, split=t1.split, copy=False, comm=t2.comm, device=t2.device, ndmin=-t1.numdims)
+            elif t1.split != t2.split:
+                # It is NOT possible to perform binary operations on tensors with different splits, e.g. split=0
```

Comment:
Have an issue documenting the fact that we need to do something here still as soon as resplit is implemented (soon done)